addons/website_sale/static/src/js/website_sale_payment.js

```diff
@@ -123,4 +123,22 @@ odoo.define('website_sale.payment', require => {
                 this.$submitButton.prop('disabled', disabledReasonFound);
             },
         }));
+
+    publicWidget.registry.WebsiteSalePayment = publicWidget.Widget.extend(
+        Object.assign({}, websiteSalePaymentMixin, {
+            selector: 'form[name="o_payment_confirm"]',
+            events: {
+                'submit': '_onSubmit',
+            },
+
+            /**
+             * Disable the confirm button when the form is submit.
+             *
+             * @private
+             * @return {undefined}
+             */
+            _onSubmit: function (ev) {
+                this.$submitButton.attr('disabled', true);
+            },
+        }));
```

Comment:
Couldn't this just be a `debounce` so that we don't need to deal with the case where the submit doesn't go through, and we then need to re-enable the button?